shared/modules/testcheck_module.py

```diff
@@ -149,9 +198,17 @@ def main():
     if len(sys.argv) != 2 or sys.argv[1].rfind('.xml') == -1:
         usage()
 
+    if not len(sys.argv) == 4:
+        schema = parse_conf_file(conf_file)
+    else:
```

Comment:
IMHO the order in which we would set the final OVAL version of the produced OVAL file for testing should follow the following scheme:
* if requested OVAL version was provided on the ```./testcheck.py``` command line, use that one,
* if not:
&nbsp; * check if the oscap version on the system supports OVAL-5.11 (similar check like in RHEL-7's Makefile we are doing right now:
&nbsp; &nbsp; [1] https://github.com/OpenSCAP/scap-security-guide/blob/master/RHEL/7/Makefile#L21
but not set that value to ```5.11``` in the ```config/oval.config``` file, but rather hardcode that version just for the next ```testcheck.py``` run. The idea is we don't want ```testcheck.py``` to hardcode the value into ```config/oval.config``` [so the content could still be properly built e.g. on RHEL-6]),
&nbsp; * if oscap on the system doesn't support OVAL-5.11 version yet, use the default value  from the config file (e.g. ```5.10```).

The point is while building OVAL-5.11 documents for RHEL-7 and Fedora now (with newer oscap versions) we can't make that version default in ```config/oval.config``` file (since need the RHEL-7 & Fedora content still to successfully built with OVAL-5.10 - older oscap currently present on RHEL-6).

Right now, having this change merged, and testing on some RHEL-7 OVAL-5.11 required check (e.g. ```service_firewalld_enabled``` from the original report:
&nbsp: [2] https://github.com/OpenSCAP/scap-security-guide/issues/765
) the following is retrieved as an output:

```
# cat /etc/redhat-release 
Red Hat Enterprise Linux Server release 7.2 Beta (Maipo)
# rpm -q openscap
openscap-1.2.5-3.el7.x86_64
# pwd
..   /scap-security-guide/RHEL/7/input/oval
# ./testcheck.py oval_5.11/service_firewalld_enabled.xml 
Evaluating with OVAL tempfile : /tmp/service_firewalld_enabledDpsO0T.xml
Writing results to : /tmp/service_firewalld_enabledDpsO0T.xml-results
File '/tmp/service_firewalld_enabledDpsO0T.xml' line 40: Element '{http://oval.mitre.org/XMLSchema/oval-definitions-5#linux}systemdunitdependency_test': This element is not expected.
File '/tmp/service_firewalld_enabledDpsO0T.xml' line 48: Element '{http://oval.mitre.org/XMLSchema/oval-definitions-5#linux}systemdunitdependency_object': This element is not expected.
File '/tmp/service_firewalld_enabledDpsO0T.xml' line 52: Element '{http://oval.mitre.org/XMLSchema/oval-definitions-5#linux}systemdunitdependency_state': This element is not expected. Expected is one of ( {http://oval.mitre.org/XMLSchema/oval-definitions-5}state, {http://oval.mitre.org/XMLSchema/oval-definitions-5#aix}interim_fix_state, {http://oval.mitre.org/XMLSchema/oval-definitions-5#aix}fileset_state, {http://oval.mitre.org/XMLSchema/oval-definitions-5#aix}fix_state, {http://oval.mitre.org/XMLSchema/oval-definitions-5#aix}no_state, {http://oval.mitre.org/XMLSchema/oval-definitions-5#aix}oslevel_state, {http://oval.mitre.org/XMLSchema/oval-definitions-5#apache}httpd_state, {http://oval.mitre.org/XMLSchema/oval-definitions-5#catos}line_state, {http://oval.mitre.org/XMLSchema/oval-definitions-5#catos}module_state, {http://oval.mitre.org/XMLSchema/oval-definitions-5#catos}version55_state ).
File '/tmp/service_firewalld_enabledDpsO0T.xml' line 19: Element '{http://oval.mitre.org/XMLSchema/oval-definitions-5}criterion': No match found for key-sequence ['oval:scap-security-guide.testing:tst:102'] of keyref '{http://oval.mitre.org/XMLSchema/oval-definitions-5}testKeyRef'.
OpenSCAP Error: Invalid OVAL Definition (5.10) content in /tmp/service_firewalld_enabledDpsO0T.xml. [oscap_source.c:205]

Error launching 'oscap' command: 
	oscap oval eval --results /tmp/service_firewalld_enabledDpsO0T.xml-results /tmp/service_firewalld_enabledDpsO0T.xml
```
This is because:
```/tmp/service_firewalld_enabledDpsO0T.xml``` contains '5.10' as the specified OVAL language version, but uses '5.11' OVAL version constructs (that's actually the point for it to be located in dedicated ```RHEL/7/input/oval/oval_5.11``` directory).

When OVAL schema version for ```/tmp/service_firewalld_enabledDpsO0T.xml``` is updated to '5.11' it works fine:
```
# cat /tmp/service_firewalld_enabledDpsO0T.xml | grep schema_version
        <oval:schema_version>5.11</oval:schema_version>
# oscap oval eval /tmp/service_firewalld_enabledDpsO0T.xml 
Definition oval:scap-security-guide.testing:def:101: true
Definition oval:scap-security-guide.testing:def:100: true
Evaluation done.
```
IOW no errors / warnings are printed out by ```oscap```.